pyxem/utils/io_utils.py

```diff
@@ -135,30 +137,621 @@ def load_hspy(filename, lazy=False, assign_to=None):
     return s
 
 
-def load_mib(filename, scan_size, sum_length=10):  # pragma: no cover
-    """Load a medipix hdr/mib file.
 
+def load_mib(mib_filename, reshape=True):
+    """Read a .mib file using dask and return as a lazy pyXem / hyperspy signal.
+    
     Parameters
     ----------
-    filename : string
-        File path and name to .hdr file.
-    scan_size : int
-        Scan size in pixels, allows the function to reshape the array into
-        the right shape.
-    sum_length : int
-        Number of lines to sum over to determine scan fly back location.
+    mib_filename : str
+        The name of the .mib file to be read.
+    
+    Returns
+    -------
+    data_hs : hyperspy.signals.Signal2D
+                The metadata adds the following domains:
+                General
+                │   └── title = 
+                └── Signal
+                    ├── binned = False
+                    ├── exposure_time = 0.001
+                    ├── flyback_times = [0.066, 0.071, 0.065, 0.017825]
+                    ├── frames_number_skipped = 90
+                    ├── scan_X = 256
+                    └── signal_type = STEM   
+        
+    
+    TODO: add data_dict as attributes to data
+        
+    """
+    hdr_stuff = parse_hdr(mib_filename)
+    data = read_mib(mib_filename, hdr_stuff)
+    exp_times_list = read_exposures(hdr_stuff, mib_filename)
+    data_dict = STEM_flag_dict(exp_times_list)
+
+    if hdr_stuff['Assembly Size'] == '2x2':
+        data = add_crosses(data)
+        
+    data_hs = hs.signals.Signal2D(data).as_lazy()
```

Comment:
We should be returning arguments in pyxem signal classes i.e. ElectronDiffraction2D and LazyElectronDiffraction2D here.

Reply:
This is now changed to:
`    data_pxm = LazyElectronDiffraction2D(data)`